packages/pingcap-ui/src/components/footer/Footer.component.js

```diff
@@ -14,13 +14,15 @@ const { useBreakpoint } = Grid;
 
 const icons = [GithubIcon, TwitterIcon, FacebookIcon, LinkedInIcon, SlackIcon, YoutubeIcon];
 
-const Footer = ({ logo, title, navItems, onNavClick }) => {
+const Footer = ({ logo, title, navItems, onNavClick, hasMargin }) => {
   const bp = useBreakpoint();
 
   const justify = bp.xs ? 'center' : undefined;
 
+  const Container = hasMargin ? Styled.ContainerWithMarginTop : Styled.Container;
```

Comment:
It's not the suggested pattern of using styled components because the container change will cause the component rerendering.

Typically, we will adapt stylings by passing a property to the styled component, like https://styled-components.com/docs/basics#adapting-based-on-props

```jsx
<Styled.Container hasMargin={hasMargin}>
  // ...
</Styled.Container
```
Then, in the `footer.styled.js`:
```javascript

export const Container = styled.div`
  background: ${colors.M2};
  ${props => props.hasMargin && css`
    margin-top: 6rem;
  `;
`;
```

Reply:
Thanks, I'm not familiar with styled component yet.

Reply:
No worries, that's the value of code review :) Practice makes perfect!